api/rest-client.ts

```diff
@@ -29,25 +44,24 @@ export interface ExtendedResponse {
 }
 
 async function requestWithRetry<T>(
-  options: AxiosRequestConfig
+  reqOptions: Options
 ): Promise<T & ExtendedResponse> {
   try {
-    const { data, headers } = await axios(options)
-
-    if (typeof data === 'object' && headers.date) {
-      data.responseTimestamp = new Date(headers.date).getTime()
+  	const options = { ...defaultOptions, ...reqOptions };
+    const { headers, body } = <{ headers: Headers, body: any }>await got(options);
+	const data = body as T & ExtendedResponse;
+    if (data !== null && typeof data === 'object' && headers.date) {
+      data!.responseTimestamp = new Date(<string>headers.date).getTime()
```

Comment:
Should be able to remove the `!` since you are checking for null
```suggestion
      data.responseTimestamp = new Date(<string>headers.date).getTime()
```